components/clarinet-deployments/src/types.rs

```diff
@@ -433,15 +478,140 @@ impl ContractPublishSpecification {
     pub contract_id: QualifiedContractIdentifier,
+    #[serde(with = "standard_principal_data_serde")]
     pub remap_sender: StandardPrincipalData,
+    #[serde(with = "remap_principals_serde")]
     pub remap_principals: BTreeMap<StandardPrincipalData, StandardPrincipalData>,
+    #[serde(with = "source_serde")]
     pub source: String,
     pub clarity_version: ClarityVersion,
     pub cost: u64,
     pub location: FileLocation,
 }
 
+pub mod source_serde {
+    use bitcoincore_rpc::jsonrpc::base64::{decode, encode};
+    use serde::{Deserialize, Deserializer, Serializer};
+    use std::str::from_utf8;
+
+    pub fn serialize<S>(x: &str, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let enc = encode(&x);
+        s.serialize_str(&enc)
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<String, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        base64_decode(&s).map_err(serde::de::Error::custom)
+    }
+
+    pub fn base64_decode(encoded: &str) -> Result<String, String> {
+        let bytes = decode(&encoded)
+            .map_err(|e| format!("unable to decode contract source: {}", e.to_string()))?;
+        let decoded = from_utf8(&bytes).map_err(|e| {
+            format!(
+                "invalid UTF-8 sequence when decoding contract source: {}",
+                e.to_string()
+            )
+        })?;
+        Ok(decoded.to_owned())
+    }
+}
+
+pub mod standard_principal_data_serde {
+    use clarity_repl::clarity::vm::types::{PrincipalData, StandardPrincipalData};
+    use serde::{Deserialize, Deserializer, Serializer};
+
+    pub fn serialize<S>(x: &StandardPrincipalData, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        s.serialize_str(&x.to_address())
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<StandardPrincipalData, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        PrincipalData::parse_standard_principal(&s).map_err(serde::de::Error::custom)
+    }
+}
+
+pub mod qualified_contract_identifier_serde {
+    use clarity_repl::clarity::vm::types::QualifiedContractIdentifier;
+    use serde::{Deserializer, Serializer};
+    use std::collections::HashMap;
+
+    pub fn serialize<'ser, S>(x: &'ser QualifiedContractIdentifier, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut m = HashMap::new();
+        m.insert("issuer", x.issuer.to_address());
+        m.insert("name", x.name.to_string());
+
+        serde::Serialize::serialize(&m, s)
+    }
+
+    pub fn deserialize<'de, D>(des: D) -> Result<QualifiedContractIdentifier, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let q: HashMap<String, String> = serde::Deserialize::deserialize(des)?;
+        let issuer = q.get("issuer").unwrap();
```

Comment:
Another unwrap we could replace with `.map_err(serde::de::Error::custom)?`. We could also maybe add some extra context here. Something like `.map_err(|e| serde::de::Error::custom(format!("qualified contract identifier deserialization requires issuer entry: {e})))?`